### PR TITLE
Profile MHV and DS Logon cred retirement messaging

### PR DESCRIPTION
**Pull request by adamwhitlock1**
## Summary

- Adds alerts around credential retirement for LOA1 users and MHV/DS Logon users when viewing the account security page.
- Uses feature toggle `profile_show_credential_retirement_messaging` and there is a corresponding BE PR to add this toggle

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/73357

## Testing done

- New UI component has unit test coverage at 100%
- E2E tests created for contact info and account security alert cases

## Screenshots

No before screenshots since these UI components didn't exist before.

**NOTE:** the original ticket/design shows an alert showing within the contact information page, but we are not going to show the alert there, because we actually do not show a 'Sign-in email' section to DS Logon / MHV users because they don't use an email address as their sign in and instead use a username that is unique to the service.

|         | Desktop | Mobile |
| ------- | ------ | ----- |
| Account Sec LOA1  | Screenshot 2024-02-27 at 10 33 46 AM | Screenshot 2024-02-27 at 10 37 18 AM |
| Account Sec DS Logon | Screenshot 2024-02-27 at 10 40 47 AM | Screenshot 2024-02-27 at 10 41 12 AM |
| Account Sec MHV user | Screenshot 2024-02-27 at 10 45 35 AM | Screenshot 2024-02-27 at 10 45 21 AM |

## What areas of the site does it impact?

Profile -> Account Security

## Acceptance criteria

- UI created and toggle-able via new feature flag name for specific user types (LOA1, MHV, and DS Logon)

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Screenshot of the developed feature is added

### CODE REVIEW

**Review by allisonlu — COMMENTED** [review]
good work!! lookin great 🎉  

a few questions:
1. are we implementing the alert without the dismissible option? (i'm comparing it to the Figma mockups)
2. LOA1 users - is the "Learn to create an account link" left out intentionally?
3. LOA1 users - should we hardcode MHV or DS logon in the alert message for LOA1 users? the mock user currently has `serviceName` set to `'idme'` so that's showing up in the alert, but i'm not sure if this will happen in prod.

Screenshot 2024-02-27 at 2 52 25 PM

**Comment by adamwhitlock1** [comment]
> good work!! lookin great 🎉
> 
> a few questions:
> 
> 1. are we implementing the alert without the dismissible option? (i'm comparing it to the Figma mockups)
> 2. LOA1 users - is the "Learn to create an account link" left out intentionally?
> 3. LOA1 users - should we hardcode MHV or DS logon in the alert message for LOA1 users? the mock user currently has `serviceName` set to `'idme'` so that's showing up in the alert, but i'm not sure if this will happen in prod.
> 
> Screenshot 2024-02-27 at 2 52 25 PM

You are so right... I had made a mental note to include session storage to make the alert dismissible.

Not sure how I missed the link in there 🤦🏼 thanks for catching that.

I could make a mock user that is MHV and DS Logon... that is probably a good idea just to have them available

**Review by micahchiang — DISMISSED** [review]

**Review by DanielMiller-agile6 — DISMISSED** [review]

**Comment by adamwhitlock1** [comment]
@allisonlu thank you for pointing out those items.

After talking to Liz, we went the route of only showing that LOA1 alert when the user in MHV/DS Logon, and showing the older alert for Id.me users.

- updated that component logic in the `AccountSecuirtyContent` component, added unit tests for that component in general since it had none, and also tested what text shows for those cases.
- added useSession storage for the dismissible alert version and made it actually dismissible now.
- lastly I added mock users to the api for loa1 mhv and dslogon account types for local testing and such

**Comment by adamwhitlock1** [comment]
@allisonlu, @DanielMiller-agile6, @micahchiang  would yall mind re-reviewing this when you have a chance?

I fixed the items that were previously mentioned, and it should be good to go now

**Review by allisonlu — APPROVED** [review]
looks great adam!! 🎉

File: src/applications/personalization/profile/components/alerts/CredentialRetirementAlerts.jsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    <>
+      <VaAlert status="continue" visible uswds>
+        <h2 slot="headline">
+          Verify your identity with Login.gov or ID.me to manage your profile
+          information
+        </h2>
+        <div>
+          <p>
+            Before we give you access to your VA.gov profile, we need to make
+            sure you’re you—and not someone pretending to be you. This helps us
```
Comment (Andrew565) [inline src/applications/personalization/profile/components/alerts/CredentialRetirementAlerts.jsx:23]: Needs space around the hyphen. "you - and"

**Review by Andrew565 — APPROVED** [review]

File: src/applications/personalization/profile/components/alerts/CredentialRetirementAlerts.jsx (hunk 1 above)
Comment (adamwhitlock1) [reply src/applications/personalization/profile/components/alerts/CredentialRetirementAlerts.jsx:23]: Pulled this content directly from Figma so it does reflect what is in the design at this time. I will follow up with designer and see what they say. Thanks @Andrew565 for reviewing!

**Pull request merged**